SmartDeviceLink/SDLTimer.m

```diff
@@ -4,6 +4,36 @@
 
 #import "SDLTimer.h"
 
+@protocol SDLTimerTargetDelegate <NSObject>
+
+- (void)timerElapsed;
+
+@end
+
+@interface SDLTimerTarget : NSObject
+
+@property (nonatomic, weak) id<SDLTimerTargetDelegate> delegate;
+
+@end
+
+@implementation SDLTimerTarget
+
+- (instancetype)initWithDelegate:(id)delegate {
+    self = [super init];
+    if (self) {
+        _delegate = delegate;
+    }
+    return self;
+}
+
+- (void)timerElapsed {
+    if ([self.delegate conformsToProtocol:@protocol(SDLTimerTargetDelegate)]) {
+        [_delegate timerElapsed];
+    }
+}
+
+@end
+
 
 @interface SDLTimer ()
```

Comment:
The `SDLTimer` needs to conform to the `SDLTimerTargetDelegate`
`@interface SDLTimer () <SDLTimerTargetDelegate>`